File: ArchitectureComponentsMVVM/app/src/main/java/com/andr/mvvm/RetofitMVVM/RetrofitViewModel/RetroViewModel.java
package com.andr.mvvm.RetofitMVVM.RetrofitViewModel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import com.andr.mvvm.RetofitMVVM.Repository.ResultModel;
import com.andr.mvvm.RetofitMVVM.Repository.WebServiceRepository;

import java.util.List;

public class RetroViewModel extends AndroidViewModel {
private final LiveData<List<ResultModel>>  retroObservable;
WebServiceRepository webServiceRepository ;
        public  RetroViewModel(Application application){
            super(application);
            webServiceRepository = new WebServiceRepository();
            retroObservable = webServiceRepository.providesWebService();
        }


    public LiveData<List<ResultModel>> getProjectRetroListObservable() {
        return retroObservable;
    }
}

File: MVVM_Kotlin/app/src/main/java/com/mvvm/kot/Kotlin_Retrofit/ViewModels/RetroViewModel.kt
package com.mvvm.kot.Kotlin_Retrofit.ViewModels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import com.mvvm.kot.Kotlin_Retrofit.Repository.APIServiceFactory
import com.mvvm.kot.Kotlin_Retrofit.Repository.ResultModel

class RetroViewModel: AndroidViewModel {
    private var retroRepository: APIServiceFactory
    private val retroObservable: LiveData<List<ResultModel>>

    constructor(application: Application): super(application){
        retroRepository = APIServiceFactory()
        retroObservable = retroRepository.providesWebService()
    }

    fun getProjectRetroListObservable(): LiveData<List<ResultModel>> {
        return retroObservable
    }

}